components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -311,6 +200,122 @@ public Response apisPost(APIDTO body, MessageContext messageContext) {
+                }
+            }
+        }
+        if (body.getAccessControlRoles() != null) {
+            String errorMessage = RestApiPublisherUtils.validateUserRoles(body.getAccessControlRoles());
+
+            if (!errorMessage.isEmpty()) {
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+        }
+        if (body.getAdditionalProperties() != null) {
+            String errorMessage = RestApiPublisherUtils
+                    .validateAdditionalProperties(body.getAdditionalProperties());
+            if (!errorMessage.isEmpty()) {
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+        }
+        if (body.getContext() == null) {
+            RestApiUtil.handleBadRequest("Parameter: \"context\" cannot be null", log);
+        } else if (body.getContext().endsWith("/")) {
+            RestApiUtil.handleBadRequest("Context cannot end with '/' character", log);
+        }
+        if (apiProvider.isApiNameWithDifferentCaseExist(body.getName())) {
+            RestApiUtil.handleBadRequest("Error occurred while adding API. API with name " + body.getName()
+                    + " already exists.", log);
+        }
+
+        //Get all existing versions of  api been adding
+        List<String> apiVersions = apiProvider.getApiVersionsMatchingApiName(body.getName(), username);
+        if (apiVersions.size() > 0) {
+            //If any previous version exists
+            for (String version : apiVersions) {
+                if (version.equalsIgnoreCase(body.getVersion())) {
+                    //If version already exists
+                    if (apiProvider.isDuplicateContextTemplate(body.getContext())) {
+                        RestApiUtil.handleResourceAlreadyExistsError("Error occurred while " +
+                                "adding the API. A duplicate API already exists for "
+                                + body.getName() + "-" + body.getVersion(), log);
+                    } else {
+                        RestApiUtil.handleBadRequest("Error occurred while adding API. API with name " +
+                                body.getName() + " already exists with different " +
+                                "context", log);
+                    }
+                }
+            }
+        } else {
+            //If no any previous version exists
+            if (apiProvider.isDuplicateContextTemplate(body.getContext())) {
+                RestApiUtil.handleBadRequest("Error occurred while adding the API. A duplicate API context " +
+                        "already exists for " + body.getContext(), log);
+            }
+        }
+
+        //Check if the user has admin permission before applying a different provider than the current user
+        String provider = body.getProvider();
+        if (!StringUtils.isBlank(provider) && !provider.equals(username)) {
+            if (!APIUtil.hasPermission(username, APIConstants.Permissions.APIM_ADMIN)) {
```

Comment:
Can't we combine the two if blocks?

Reply:
Can. But in here else block assign the value of username to the provider name in case of the provider getting null or empty value. The two if blocks have been used to preserve the logic